src/org/ods/orchestration/FinalizeStage.groovy

```diff
@@ -140,42 +140,52 @@ class FinalizeStage extends Stage {
         def repoPushTasks = [ : ]
         def repoSize = flattenedRepos.size()
+        steps.echo "Finalize: Push flattenede repos"
+        steps.echo "$flattenedRepos"
         for (def i = 0; i < repoSize; i++) {
             def repo = flattenedRepos[i]
-            repoPushTasks << [ (repo.id): {
-                steps.dir("${steps.env.WORKSPACE}/${MROPipelineUtil.REPOS_BASE_DIR}/${repo.id}") {
-                    if (project.isWorkInProgress) {
-                        String branchName = repo.data.git.branch ?: repo.branch
-                        git.pushRef(branchName)
-                    } else if (project.isAssembleMode) {
-                        if (!git.remoteTagExists(project.targetTag)) {
-                            git.createTag(project.targetTag)
-                        }
-                        git.pushBranchWithTags(project.gitReleaseBranch)
-                    } else {
-                        if (!git.remoteTagExists(project.targetTag)) {
-                            git.createTag(project.targetTag)
-                        }
-                        git.pushRef(project.targetTag)
-                    }
-                }
+            if (repo.include) {
+                repoPushTasks << repoClosureMap(repo, steps, git)
             }
-            ]
         }
         repoPushTasks.failFast = true
         script.parallel(repoPushTasks)
     }
 
+    private Map<Object, Closure> repoClosureMap(repo, steps, git) {
+        return [(repo.id): {
+            steps.dir("${steps.env.WORKSPACE}/${MROPipelineUtil.REPOS_BASE_DIR}/${repo.id}") {
+                if (project.isWorkInProgress) {
+                    String branchName = repo.data.git.branch ?: repo.branch
+                    git.pushRef(branchName)
+                } else if (project.isAssembleMode) {
+                    if (!git.remoteTagExists(project.targetTag)) {
+                        git.createTag(project.targetTag)
+                    }
+                    git.pushBranchWithTags(project.gitReleaseBranch)
+                } else {
+                    if (!git.remoteTagExists(project.targetTag)) {
+                        git.createTag(project.targetTag)
+                    }
+                    git.pushRef(project.targetTag)
+                }
+            }
+        }
+        ]
+    }
+
     private void gatherCreatedExecutionCommits(IPipelineSteps steps, GitService git) {
         def flattenedRepos = repos.flatten()
         def gatherCommitTasks = [ : ]
         def repoSize = flattenedRepos.size()
         for (def i = 0; i < repoSize; i++) {
             def repo = flattenedRepos[i]
             gatherCommitTasks << [ (repo.id): {
-                steps.dir("${steps.env.WORKSPACE}/${MROPipelineUtil.REPOS_BASE_DIR}/${repo.id}") {
-                    repo.data.git.createdExecutionCommit = git.commitSha
-                    steps.echo "repo.id: ${repo.id}: ${repo.data.git.createdExecutionCommit}"
+                if (repo.include) {
```

Comment:
Wouldn't it be possible to not add the entry to gatherCommitTasks at all instead of adding an empty closure?